lib/active_merchant/billing/gateways/priority.rb

```diff
@@ -161,21 +161,22 @@ def exp_date(credit_card)
       end
 
       def add_type_merchant_purchase(params, merchant, is_settle_funds, options)
-        params['cardPresent'] = false
-        params['cardPresentType'] = 'CardNotPresent'
-        params['isAuth'] = true
+        params['cardPresent'] = options[:card_present]? options[:card_present] : 'false'
+        params['cardPresentType'] = options[:card_present_type]? options[:card_present_type] : 'CardNotPresent'
+        params['isAuth'] = options[:is_auth]? options[:is_auth] : 'true'
         params['isSettleFunds'] = is_settle_funds
         params['isTicket'] = false
         params['merchantId'] = merchant
         params['mxAdvantageEnabled'] = false
-        params['paymentType'] = 'Sale'
+        params['paymentType'] = options[:payment_type]? options[:payment_type] : 'Sale'
+        add_purchases_data(params,options[:purchases]) if options[:purchases]
         params['purchases'] = options[:purchases]
```

Comment:
You can remove this line now that you have `add_purchases_data`

Reply:
Done